server/series/api.py

```diff
@@ -0,0 +1,436 @@
+        return 400, {"message": "Series does not exist"}
+
+    try:
+        team = Team.objects.get(slug=team_slug)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    try:
+        player = Player.objects.get(user=request.user)
+    except Player.DoesNotExist:
+        return 400, {"message": "Player does not exist"}
+
+    return 200, SeriesRosterInvitation.objects.filter(series=series, team=team, to_player=player)
+
+
+@router.get(
+    "/{series_slug}/team/{team_slug}/invitations-sent",
+    response={200: list[SeriesRosterInvitationSchema], 400: Response, 401: Response},
+)
+def series_team_roster_invitations_sent(
+    request: AuthenticatedHttpRequest, series_slug: str, team_slug: str
+) -> tuple[int, QuerySet[SeriesRosterInvitation] | message_response]:
+    try:
+        series = Series.objects.get(slug=series_slug)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    try:
+        team = Team.objects.get(slug=team_slug)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    if request.user not in team.admins.all():
+        return 401, {"message": "Only team admins can see all the invitiations they've sent"}
+
+    return 200, SeriesRosterInvitation.objects.filter(
+        series=series, team=team, from_user=request.user
+    )
+
+
+@router.get(
+    "/{series_slug}/team/{team_slug}/roster",
+    response={200: list[SeriesTeamRosterSchema], 400: Response, 401: Response},
+)
+def get_team_series_roster(
+    request: AuthenticatedHttpRequest, series_slug: str, team_slug: str
+) -> tuple[int, Any | message_response]:
+    try:
+        series = Series.objects.get(slug=series_slug)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    try:
+        team = Team.objects.get(slug=team_slug)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    return 200, SeriesRegistration.objects.filter(series=series, team=team).order_by(
+        "player__user__first_name"
+    )
+
+
+@router.put(
+    "/{series_id}/team/{team_id}/roster",
+    response={200: SeriesRegistrationSchema, 400: Response, 401: Response},
+)
+def add_myself_to_team_series_roster(
+    request: AuthenticatedHttpRequest,
+    series_id: int,
+    team_id: int,
+    player_details: SeriesRegistrationAddSchema,
+) -> tuple[int, SeriesRegistration | message_response]:
+    try:
+        series = Series.objects.get(id=series_id)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    try:
+        team = Team.objects.get(id=team_id)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    if request.user not in team.admins.all():
+        return 401, {"message": "Only team admins can add themselves to a series roster"}
+
+    try:
+        player = Player.objects.get(id=player_details.player_id)
```

Comment:
why cant we get this from request.user?

Reply:
why do we even need player info sent?

Reply:
yes yes, changing this as we speak. meant to change and push this

Reply:
changed to remove sending player info